src/odemis/gui/comp/viewport.py

```diff
@@ -1065,7 +1066,175 @@ def _on_peak_method(self, state):
         pass
 
 
-class PointSpectrumViewport(PlotViewport):
+class NavigablePlotViewport(PlotViewport):
+    """ Abstract Class for displaying plotted data """
+    __metaclass__ = no_conflict.classmaker(right_metas=(ABCMeta,))
+
+    # Default class
+    canvas_class = miccanvas.NavigableBarPlotCanvas
+    bottom_legend_class = AxisLegend
+    left_legend_class = AxisLegend
+
+    def __init__(self, *args, **kwargs):
+        super(NavigablePlotViewport, self).__init__(*args, **kwargs)
+
+        self.hrange = model.TupleVA()
+        self.bottom_legend.tick_spacing = 80
+        self.hrange.subscribe(self.onHRange)
+
+        self.vrange = model.TupleVA()
+        self.vrange.subscribe(self.onVRange)
+
+        # Indicate a left mouse button drag in the canvas
+        # Note: *only* use it indicate that the *canvas* is performing an operation related to
+        # dragging!
+        self._hdragging = False
+        self._hdrag_scale_factor = -0.1
+
+        # The amount of pixels shifted in the current drag event
+        self.drag_shift = (0, 0)  # px, px
+        #  initial position of mouse when started dragging
+        self.drag_init_pos = (0, 0)  # px, px
+        self._last_pos = (0, 0)
+        self._last_time = 0
+
+        # Track if the legend was dragged. It should be reset to False at the
+        # end of button up handlers, giving overlay the chance to check if a
+        # drag occurred.
+        self.was_dragged = False
+
+        self.canvas.Bind(wx.EVT_MOUSEWHEEL, self.on_hlegend_scroll)
+        self.canvas.Bind(wx.EVT_MIDDLE_DOWN, self.on_hlegend_drag_start)
+        self.canvas.Bind(wx.EVT_MIDDLE_UP, self.on_hlegend_drag_end)
+        self.canvas.Bind(wx.EVT_MOTION, self.on_hlegend_motion)
+        self.bottom_legend.Bind(wx.EVT_MOUSEWHEEL, self.on_hlegend_scroll)
+        self.left_legend.Bind(wx.EVT_MOUSEWHEEL, self.on_vlegend_scroll)
```

Comment:
Also need to handle drag on the left legend, or is it already handle somewhere else?